wellcomeml/ml/bert_semantic_equivalence.py

```diff
@@ -0,0 +1,176 @@
+import os
+
+import tensorflow as tf
+
+from transformers import (
+    BertConfig,
+    BertTokenizer,
+    TFBertForSequenceClassification,
+    glue_convert_examples_to_features,
+    glue_processors,
+)
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.model_selection import train_test_split
+
+from wellcomeml.ml.bert_vectorizer import check_cache_and_download
+
+
+class SemanticEquivalenceClassifier(BaseEstimator, TransformerMixin):
+    """
+    Class to fine-tune BERT-type models for semantic equivalence, for example
+    paraphrase, textual similarity and other NLU tasks
+    """
+    def __init__(self, pretrained='bert',
+                 batch_size=32,
+                 eval_batch_size=32*2,
+                 learning_rate=3e-5,
+                 test_size=0.2,
+                 max_length=128):
+        self.pretrained = pretrained
+        self.batch_size = batch_size
+        self.eval_batch_size = eval_batch_size
+        self.learning_rate = learning_rate
+        self.test_size = test_size
+        self.max_length = max_length
+
+        # Defines attributes that will be initialised later
+        self.config = None
+        self.tokenizer = None
+        self.model = None
+        self.train_dataset = None
+        self.valid_dataset = None
+        self.train_steps = None
+        self.valid_steps = None
+
+    # Bert models have a tensorflow checkopoint, otherwise,
+    # we need to load the pytorch versions with the parameter `from_pt=True`
+
+    def _initialise_models(self):
+        if self.pretrained == 'bert':
+            model_name = 'bert-base-cased'
+            from_pt = False
+        elif self.pretrained == 'scibert':
+            model_name = check_cache_and_download('scibert_scivocab_cased')
+            from_pt = True
+
+        print(model_name)
```

Comment:
logger maybe better?

Reply:
Yes this slipped through the code

Reply:
Fixed in https://github.com/wellcometrust/WellcomeML/pull/43/commits/cca45683f07aedc492664e29db13477ab91284eb, https://github.com/wellcometrust/WellcomeML/pull/43/commits/ef399baaf9db8c67c20c441e35f6b74cf88c0f9f